Sources/Kitura/Router.swift

```diff
@@ -492,11 +548,16 @@ extension Router : ServerDelegate {
     /// - Parameter response: The `ServerResponse` object used to send responses to the
     ///                      HTTP request at the [Kitura-net](http://ibm-swift.github.io/Kitura-net/) API level.
     public func handle(request: ServerRequest, response: ServerResponse) {
-        let routeReq = RouterRequest(request: request)
+        var decoder: (() -> BodyDecoder)?
+        if let mediaType = MediaType(headers: request.headers) {
```

Comment:
We should test this to see how much overhead it adds for requests such as a simple `GET` that does not contain a content type (or body).